crates/metal-codegen-llvm/src/safeties.rs

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+//! Safe Rust abstractions for several LLVM types
+
+use std::slice;
+
+use llvm_sys::{
+    core::{LLVMDisposeMemoryBuffer, LLVMGetBufferSize, LLVMGetBufferStart},
+    prelude::LLVMMemoryBufferRef,
+};
+
+/// Safe abstraction over LLVM's MemoryBuffer
+pub struct MemoryBuffer {
+    llvm_buf: LLVMMemoryBufferRef,
+}
+
+impl MemoryBuffer {
+    pub fn new(buf: LLVMMemoryBufferRef) -> Self {
+        Self { llvm_buf: buf }
+    }
+
+    /// Turns an LLVM buffer into a Vec<u8>
+    #[inline]
+    pub fn to_vec(&self) -> Vec<u8> {
+        unsafe {
+            let start = LLVMGetBufferStart(self.llvm_buf) as *const u8;
+            let size = LLVMGetBufferSize(self.llvm_buf) as usize;
+            slice::from_raw_parts(start, size).to_vec()
+        }
+    }
+}
+
+impl Drop for MemoryBuffer {
```

Comment:
Same question as for the other Drop impl

Reply:
> Same question as for the other Drop impl

Just prevents forgetting to dispose of the LLVM MemoryBuffer (and also just keeps code clean so we don't need to have an `LLVMDisposeMemoryBuffer` every other line)

Reply:
> What "not disposing of any LLVM references/types" results in?